01_stratified_classification/nbconverted/lasso_range_all_optimizers.py

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments, summary across all genes
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.colors import Normalize
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+ll_results_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'results', 'optimizer_compare_ll', 'gene'
+)
+
+sgd_results_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'results', 'optimizer_compare_sgd', 'gene'
+)
+
+plot_gene = 'EGFR'
+metric = 'aupr'
+
+output_plots = True
+output_plots_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'optimizers_plots'
+)
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+ll_nz_coefs_df = []
+
+# get coefficient info for training dataset specified above
+for coef_info in au.generate_nonzero_coefficients_lasso_range(ll_results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        ll_nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+ll_nz_coefs_df = pd.DataFrame(
+    ll_nz_coefs_df,
+    columns=['gene', 'signal', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+ll_nz_coefs_df.lasso_param = ll_nz_coefs_df.lasso_param.astype(float)
+
+print(ll_nz_coefs_df.gene.unique())
```

Comment:
Are these pre-selected by you or are these the actual non-zero genes that were picked by the trained model? If the latter, it's pretty impressive that all the "usual suspects" are there.

Reply:
Nope, it's the Vogelstein genes, I'm an idiot.

Reply:
It is! I'll make a note of this though, "all genes" is pretty ambiguous